text/0118-search-api.md

```diff
@@ -108,156 +111,273 @@ It returns the number of documents matching the current search query for each gi
+
+##### 3.1.1.13.1. Applying `cropMarker`
+
+###### 3.1.1.13.1.1. Matched Part To Be Cropped
+
+The cropping algorithm tries to match the window with the highest density of query terms within the `cropLength` limit. Then it will pick the window that contains the more ordered query terms.
+
+If two windows have the same density, it chooses the first one within the attribute to be cropped. It also means that only one cropped part is returned.
+
+If no match is found when selecting a part to be cropped. The returned value in `_formatted` will start at the beginning of the attribute until the distance of `cropLength` without any `cropMarker` being applied.
+
+###### 3.1.1.13.1.2. Positioning Markers
+
+If the cropped part has been matched against query terms and contains the beginning of the attribute to be cropped, the `cropMarker` is not placed to the left of the cropped part.
+
+If the cropped part has been matched against query terms and contains the end of the attribute to be cropped, the `cropMarker` is not placed to the right of the cropped part.
+
+#### 3.1.1.14. `matches`
 
 - Type: Boolean
 - Required: False
 - Default: `false`
 
-Adds a `_matchesInfo` object to the search response that contains the location of each occurrence of queried terms across all fields. This is useful when more control is needed than offered by the built-in highlighting/cropping features.
+Adds a `_matchesInfo` object to the search response that contains the location of each occurrence of queried terms across all fields. It's useful when more control is needed than offered by the built-in highlighting/cropping features.
 
-- 🔴 Sending a value with a different type than `Boolean` or `null` for `matches` will return a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+- 🔴 Sending a value with a different type than `Boolean` or `null` for `matches` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
 
-#### 1.2.2. Search response
+### 3.2. Search Response Properties
 
-| Field                   | Type                         | Required |
-|-------------------------|------------------------------|----------|
-| hits                    | Array[Hit]                   | True     |
-| limit                   | Integer                      | True     |
-| offset                  | Integer                      | True     |
-| nbHits                  | Integer                      | True     |
-| exhaustiveNbHits        | Boolean                      | True     |
-| facetsDistribution      | Object                       | False    |
-| exhaustiveFacetsCount   | Boolean                      | False    |
-| processingTimeMs        | Integer                      | True     |
-| query                   | String                       | True     |
+| Field                                                 | Type                         | Required |
```

Comment:
What does it mean "required" for a search response?
Do you mean "present by default"?

Reply:
Yes, that's right!